test/suite/test_live_restore02.py

```diff
@@ -0,0 +1,109 @@
+# Enable background thread migration and loop until it completes
+@wttest.skip_for_hook("tiered", "using multiple WT homes")
+class test_live_restore02(wttest.WiredTigerTestCase):
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('row_integer', dict(key_format='i', value_format='S')),
+    ]
+    scenarios = make_scenarios(format_values)
+    nrows = 10000
+
+    def get_stat(self, statistic):
+        stat_cursor = self.session.open_cursor("statistics:")
+        val = stat_cursor[statistic][2]
+        stat_cursor.close()
+        return val
+
+    def test_live_restore02(self):
+        # Live restore is not supported on Windows.
+        if os.name == 'nt':
+            return
+
+        uris = ['file:foo', 'file:bar', 'file:cat']
+        # Create a data set with a 3 collections to restore on restart.
+        # Populate 3 collections
+        ds0 = SimpleDataSet(self, uris[0], self.nrows,
+          key_format=self.key_format, value_format=self.value_format)
+        ds0.populate()
+        ds1 = SimpleDataSet(self, uris[1], self.nrows,
+          key_format=self.key_format, value_format=self.value_format)
+        ds1.populate()
+        ds2 = SimpleDataSet(self, uris[2], self.nrows,
+          key_format=self.key_format, value_format=self.value_format)
+        ds2.populate()
+
+        # Close the default connection.
+        self.close_conn()
+
+        copy_wiredtiger_home(self, '.', "SOURCE")
+
+        # Remove everything but SOURCE / stderr / stdout.
+        for f in glob.glob("*"):
+            if not f == "SOURCE" and not f == "stderr.txt" and not f == "stdout.txt":
+                os.remove(f)
+
+        self.open_conn(config="statistics=(all),live_restore=(enabled=true,path=\"SOURCE\",threads_max=1)")
+
+        state = 0
+        timeout = 120
+        iteration_count = 0
+        # Build in a 2 minute timeout. Once we see the complete state exit the loop.
+        while (iteration_count < timeout):
+            state = self.get_stat(stat.conn.live_restore_state)
+            self.pr("Looping until finish, live restore state is: " + str(state))
+            if (state == 2):
+                break
+            time.sleep(1)
+        self.assertEquals(state, 2)
+
+        conn2 = self.setUpConnectionOpen('SOURCE/')
+        session2 = self.setUpSessionOpen(conn2)
+        session = self.setUpSessionOpen(self.conn)
+        # Validate that the collections in source match ours in the destination.
+        for uri in uris:
+            cursor = session.open_cursor(uri)
+            cursor2 = session2.open_cursor(uri)
+            while True:
+                ret = cursor.next()
+                ret2 = cursor2.next()
+                if ret != 0:
+                    self.assertEquals(ret, wiredtiger.WT_NOTFOUND)
+                    self.assertEquals(ret2, wiredtiger.WT_NOTFOUND)
+                    break
+                self.assertEquals(cursor.get_key(), cursor2.get_key())
```

Comment:
I see this being useful for future live_restore tests. Can we make it a helper function.

Reply:
In the interest of getting this merged so people are unblocked can we push that to a separate ticket?

Reply:
sure thing